source/features/clone-branch.tsx

```diff
@@ -0,0 +1,83 @@
+import gitBranch from 'octicon/git-branch.svg';
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate, {DelegateEvent} from 'delegate-it';
+import * as api from '../libs/api';
+import features from '../libs/features';
+import {getRepoURL} from '../libs/utils';
+
+async function cloneBranch(event: DelegateEvent<MouseEvent, HTMLAnchorElement>): Promise<void|false> {
+	const currentTarget = event.delegateTarget;
+	const branchElement = (currentTarget.closest('[data-branch-name]') as HTMLAnchorElement);
+	const newBranchName = prompt('Enter the new branch name')?.trim();
+	if (!newBranchName) {
+		return false;
+	}
+
+	// eslint-disable-next-line no-control-regex
+	const invalidBranchName = new RegExp(/^[./]|\.\.|@{|[/.]$|^@$|[~^:\u0000-\u0020\u007F\s?*]/);
+	if (invalidBranchName.test(newBranchName)) {
+		alert(`'${newBranchName}' contains an invalid branch name character \n see https://git-scm.com/docs/git-check-ref-format for more details`);
+		return false;
+	}
```

Comment:
Ideally you'd use a loop with `prompt` until it's valid or empty/cancelled:

```js
let newBranchName = prompt('Enter the new branch name')?.trim();

while(invalidBranchName.test(newBranchName))) {
	newBranchName = prompt('Unsupported filename (https://git-scm.com/docs/git-check-ref-format) \nEnter the new branch name', branchName)?.trim();
}

if (!branchName) {
	return;
}
```

Reply:
To tell you the truth I did not even think you would want the check to begin with. I added it very quickly just to get your feedback. I will go through it later.
I came down with bronchitis so I hope I can do it today.